### PR TITLE
feat: add x509 support

### PR DESCRIPTION
**Pull request by ajpauwels**

### CODE REVIEW

**Comment by capt-haddock** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **ajpauwels**
You can assign the PR to them by writing `/assign @ajpauwels` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=pauwels-labs%2Fredact-crypto).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/pauwels-labs/redact-crypto/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by capt-haddock** [comment]
@ajpauwels: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-test | 92193fbf7da1a511c7fece925804f9082a2b1589 | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/abs-redact-crypto-pr-5-unit-test-vdnjb) | `/test unit-test`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: src/key/sodiumoxide.rs
Comment (howhelan) [inline src/key/sodiumoxide.rs:751]: curious about the difference between these two?

**Review by howhelan — APPROVED** [review]

**Comment by capt-haddock** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[howhelan](https://github.com/pauwels-labs/redact-crypto/pull/5#pullrequestreview-749122434 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=pauwels-labs%2Fredact-crypto).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/pauwels-labs/redact-crypto/blob/main/OWNERS)~~ [howhelan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
